src/CloudNative.CloudEvents.Avro/BasicGenericRecordSerializer.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Avro;
+using Avro.Generic;
+using Avro.IO;
+using CloudNative.CloudEvents.Avro.Interfaces;
+using System;
+using System.IO;
+
+namespace CloudNative.CloudEvents.Avro;
+
+/// <summary>
+/// The default implementation of the <see cref="IGenericRecordSerializer"/>.
+/// </summary>
+/// <remarks>
+/// Makes use of the Avro <see cref="DefaultReader"/> and <see cref="DefaultWriter"/>
+/// together with the embedded Avro schema.
+/// </remarks>
+sealed internal class BasicGenericRecordSerializer : IGenericRecordSerializer
+{
+    private readonly DefaultReader avroReader;
+    private readonly DefaultWriter avroWriter;
+
+    public BasicGenericRecordSerializer(RecordSchema avroSchema)
```

Comment:
If the schema is needed for this, presumably it would be needed (or at least useful) for other real implementations too?
If the schema is immutable (which it looks like it is), let's expose it via a public static property in the event formatter.

Reply:
Yes, I agree. Great suggestion, that may be useful for others.

Reply:
Thanks for taking the time to review this so soon, it's greatly appreciated